tests/ui/empty_docs.rs

```diff
@@ -0,0 +1,27 @@
+#![allow(unused)]
+#![warn(clippy::empty_docs)]
+
+pub mod outer_module {
+
+    //!
+
+    //! valid doc comment
+
+    //!!
+
+    //!! valid doc comment
+
+    ///
+
+    /// valid doc comment
+
+    /**
+     *
+     */
+
+    /**
+     * valid block doc comment
+     */
+
+    pub mod inner_module {}
```

Comment:
None of these are actually empty documentation, this is ~equivalent to

```rust
//!
//! valid doc comment
//!!
//!! valid doc comment

///
/// valid doc comment
/// 
/// valid block doc comment
```

Because doc comments and `#[doc]` attributes are merged together

This really should go in `docs.rs` where this merging/parsing is all handled already

Reply:
@Alexendoo according to https://doc.rust-lang.org/reference/comments.html there are one line and block doc comments.
What you show in your example is 8 separate one-line doc comments. They are not being merged in 2, just because there are no line breaks within them. So lint in this PR will trigger on lines 1, 3, 5, 6 (and I'm updating the test to demonstrate this).

As for block doc comments, I assume they might have empty lines (for formatting purposes) because the definition of `empty` for a block comment is when all lines are empty.

Tbh I do not understand the part about merging with `#[doc]` attributes. Can you please explain?

Reply:
```
This really should go in doc.rs where this merging/parsing is all handled already
```
there is no much merging/parsing happening in this current lint. The idea is to have all doc-related lints in doc.rs?

Reply:
I'm not saying the attributes are merged in the AST/HIR, but the multiple attributes representing doc comments are merged into a single piece of documentation by rustdoc, be they line comments, block comments or [`#[doc = ".."]` attributes](https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html)

My example was to represent what rustdoc approximately sees as the documentation, not to say that doc comments get converted to line comments. A screenshot may be clearer:

```rust
    ///

    /// valid doc comment

    /**
     *
     */

    /**
     * valid block doc comment
     */

    pub mod inner_module {}
```

![image](https://github.com/rust-lang/rust-clippy/assets/1830331/4c16758f-b77e-4c48-ad27-1bb5460b95a9)

The whitespace between the doc comments are not significant, e.g.


```rust
/// a

///

/// b
```

is the same as

```rust
/// a
///
/// b
```

But we certainly don't want to call the middle line comment empty, it's significant because it causes `a` and `b` to be separate paragraphs

Reply:
> there is no much merging/parsing happening in this current lint. The idea is to have all doc-related lints in doc.rs?

Yeah, it could go here

https://github.com/rust-lang/rust-clippy/blob/3662bd8f587c9305e4d231a9456e63b92b8a7729/clippy_lints/src/doc.rs#L493-L495

With `.is_empty()` being changed to check that `doc` only contains whitespace if needed, you can use https://doc.rust-lang.org/nightly/nightly-rustc/rustc_resolve/rustdoc/fn.span_of_fragments.html on `fragments` to get the span